### DISCUSSION TITLE
What is the logic of variations in a listing

### DISCUSSION THREAD

**Discussion by tatlicak** [etsy/open-api#1081] [Q&A]
Hi,
I am developing a web application with Etsy API v3. I managed to perform the listing creation successfully. 
However, I couldn't fully grasp the logic of adding variations to a listing. 
I need assistance from someone who can help me understand how to add variations to an existing or newly created listing through the API and which endpoints to use. 
You might ask me to check the documentation, but I've already gone through it, and I'm still confused.  I could find only this endpoints about variations.
**updateVariationImages
getListingVariationImages
uploadListingFile**
So, if anyone can help, I would appreciate it. 
Thanks

**Comment by emergingdzns** [comment]
@tatlicak What you need to do after creating a listing is then call the [getListingInventory](https://developers.etsy.com/documentation/reference#operation/getListingInventory) endpoint. This will give you the basic single-product inventory.

Once you have that, you can begin adding variations. It's hard to explain but the tutorial is actually pretty good:
https://developers.etsy.com/documentation/tutorials/listings#updating-inventory

The main thing is to make sure that if the variations you have don't affect price, then make sure that the prices all match. Same goes for the skus and quantity. But if any of those variations do affect any of those, you have to put the property ids into the relevant variable arrays at the end: price_on_property, quantity_on_property, sku_on_property.

Not sure what language you are using but the PHP Curl examples in the tutorial do show a decent set of data. The trick is in knowing the property ids. Those you can get from the taxonomy endpoints. I recommend you query those and store the data. The taxonomy rarely changes, so you can update periodically and not hit them every time you create a listing.

Hope it helps.

**Reply by tatlicak** [reply]
Thanks for reply @emergingdzns
